File: app/src/main/java/com/stevenfrew/beatprompter/filter/SetListFilter.java
package com.stevenfrew.beatprompter.filter;

import com.stevenfrew.beatprompter.cache.SongFile;

import java.util.List;

public class SetListFilter extends SongFilter
{
    SetListFilter(String name,List<SongFile> songs)
    {
        super(name,songs,false);
    }

    public boolean containsSong(SongFile sf)
    {
        return mSongs.contains(sf);
    }

    public boolean equals(Object o) {
        return o instanceof SetListFilter && mName.equals(((SetListFilter) o).mName);
    }
}

File: app/src/main/java/com/stevenfrew/beatprompter/filter/SetListFilter.kt
package com.stevenfrew.beatprompter.filter

import com.stevenfrew.beatprompter.cache.SongFile

open class SetListFilter internal constructor(name: String, songs: MutableList<SongFile>) : SongFilter(name, songs, false) {

    fun containsSong(sf: SongFile): Boolean {
        return mSongs.contains(sf)
    }

    override fun equals(other: Any?): Boolean {
        return other is SetListFilter && mName == other.mName
    }
}